### ISSUE TITLE
Mutter(?) causes weird artifacts/ghost windowborders

### ISSUE THREAD

**Issue by J0Ar** [elementary/os-patches#57]
elementaryOS has some strange visual glitches since some time now…1 or 2 months i think.
Look at the [Video](https://streamable.com/stu1m)
So far these are visible in Firefox and Synaptic's context menus.

System:![Bildschirmfoto von 2019-07-15 02-22-59](https://user-images.githubusercontent.com/23266409/61191188-7f7a9500-a6a7-11e9-9b45-2f804d36fa71.png)

I think this has something to do with Mutter…i tried the LiveSystem with the older packages…
- Boot USB 'elementaryos-5.0-stable.20181016.iso'
- changed resolution from 1920x1080 to 1600x900 as normal and activated WLAN
- disabled the Repo 'dep cdrom:…'  with 'sudo nano /etc/apt/sources.list'
- 'sudo apt update && sudo apt install firefox'
- used my own firefox user profil because the bookmarks toolbar is more full^^
- **Result: Everything is fine**

Then:
- installed the newest mutter package 'sudo apt install libmutter-2-0' (+ dependencies)
- logged out…in
- **Result: Artifacts/ghost window borders are visible**

So i think this has nothing to do with odd drivers or kernels…or am i wrong?

**Comment by peteruithoven** [comment]
For next time I'd recommend taking screen recordings with [Screenrecorder](https://appcenter.elementary.io/com.github.mohelm97.screenrecorder). 

On a fresh image installing mutter shouldn't do anything since it's already installed as far as I know?
Try:
```
$ apt search libmutter-2.0
Sorting... Done
Full Text Search... Done
libmutter-2-0/bionic,now 3.28.4-0ubuntu18.04.1+elementary3~ubuntu5.0.1 amd64 [installed,automatic]
  window manager library from the Mutter window manager
```

The glitch that's shown in the video sounds like something Firefox might be able to fix. 
Maybe search on their on their issue list: https://bugzilla.mozilla.org/buglist.cgi?quicksearch=firefox+context+menu ?

You mention it also happens in Synaptic's context menus, which make it sound like a general Gtk issue? In that case other people, on for example Ubuntu should experience the same issue.

**Comment by J0Ar** [comment]
First I tried with Screenrecorder...but when Screenrecorder is recording...these artifacts are gone!? Don't ask me why 🤷🏻‍♂️ Maybe I should try this again to confirm this once again. 
What do you mean by 'mutter should do anything'? In a fresh install or liveCD the version of mutter is an older one (3.28.2...idk the correct one, also I'm at work at the moment) but with this older version there are no glitches visible. 
This is the reason, why I think it has something to do with an update between last year and now. 
I will investigate later when I'm at home. 🙃
Thanks for the answer...

**Comment by peteruithoven** [comment]
You're right, after a `apt update`, `apt install libmutter-2-0` does upgrade mutter.
```
$ sudo apt install libmutter-2-0
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  mutter-common
The following packages will be upgraded:
  libmutter-2-0 mutter-common
```

When you don't upgrade mutter and re-login, you don't see the artifacts? 

Checking the mutter issue queue would probably be interesting: 
https://gitlab.gnome.org/GNOME/mutter/issues

**Comment by J0Ar** [comment]
Yes...this is what I have done with the liveCD.

First: tried Firefox + original mutter package of the last stable liveCD >>> Fine!

Second: tried Firefox + updated mutter packages >>> glitches!!! 
The 'logout > login' is nessecary to get the upgraded mutter working.
